### DISCUSSION TITLE
Run with Restart file

### DISCUSSION THREAD

**Discussion by PGhaneei** [NCAR/hrldas#221] [Q&A] [answered]
Hello, 

I am using HRLDAS to run the Noah-MP model over the CONUS domain using NLDAS forcing. I am running the model for a single day, saving the restart file, and using it to run the subsequent day.

When I compare the states that I obtain (soil moisture: SMC) with those of a continuous run (without using restart files), I notice the states are different. 
Please note that, I am using the same HRLDAS setup file for all runs with restart files.

I would greatly appreciate any insights into the potential issue, as I believe the states should be the same for both experiments.

Below is the configuration file that I update daily and use to run the next day with the restart file:

&NOAHLSM_OFFLINE

 HRLDAS_SETUP_FILE = "/hrldas/NLDAS_forcing/LDASIN_2015/HRLDAS_setup_2015040123_d1"
 INDIR = "hrldas/NLDAS_forcing/LDASIN_2015_FullYear/0/"
 OUTDIR = "hrldas/run/1/Outputs/0/"

 START_YEAR  = 2015
 START_MONTH  = 04
 START_DAY  = 04
 START_HOUR  = 00
 START_MIN   = 00

 RESTART_FILENAME_REQUESTED = "hrldas/run/1/0/RESTART.2015040400_DOMAIN1"

 KDAY = 1
 SPINUP_LOOPS = 0

 FORCING_NAME_T = "T2D"
 FORCING_NAME_Q = "Q2D"
 FORCING_NAME_U = "U2D"
 FORCING_NAME_V = "V2D"
 FORCING_NAME_P = "PSFC"
 FORCING_NAME_LW = "LWDOWN"
 FORCING_NAME_SW = "SWDOWN"
 FORCING_NAME_PR = "RAINRATE"

 DYNAMIC_VEG_OPTION                = 6
 CANOPY_STOMATAL_RESISTANCE_OPTION = 1
 BTR_OPTION                        = 1
 SURFACE_RUNOFF_OPTION             = 1
 SUBSURFACE_RUNOFF_OPTION          = 1
 DVIC_INFILTRATION_OPTION          = 1
 SURFACE_DRAG_OPTION               = 1
 FROZEN_SOIL_OPTION                = 1
 SUPERCOOLED_WATER_OPTION          = 1
 RADIATIVE_TRANSFER_OPTION         = 3
 SNOW_ALBEDO_OPTION                = 2
 PCP_PARTITION_OPTION              = 1
 SNOW_THERMAL_CONDUCTIVITY         = 1
 TBOT_OPTION                       = 2
 TEMP_TIME_SCHEME_OPTION           = 1
 GLACIER_OPTION                    = 1
 SURFACE_RESISTANCE_OPTION         = 1
 SOIL_DATA_OPTION                  = 1
 PEDOTRANSFER_OPTION               = 1
 CROP_OPTION                       = 1
 IRRIGATION_OPTION                 = 0
 IRRIGATION_METHOD                 = 0
 TILE_DRAINAGE_OPTION              = 0

 FORCING_TIMESTEP = 3600
 NOAH_TIMESTEP    = 1800
 OUTPUT_TIMESTEP  = 86400

 SPLIT_OUTPUT_COUNT = 1
 SKIP_FIRST_OUTPUT = .false.
 RESTART_FREQUENCY_HOURS = 24

 ! XSTART = 20
 ! XEND   = 20
 ! YSTART = 203
 ! YEND   = 203

 NSOIL=4
 soil_thick_input(1) = 0.10
 soil_thick_input(2) = 0.40
 soil_thick_input(3) = 0.60
 soil_thick_input(4) = 1.00

 ZLVL = 10.0

 SF_URBAN_PHYSICS = 0
 USE_WUDAPT_LCZ = 0
 
/

**Comment by cenlinhe** [comment]
Which day and hour of the SMC are you comparing? The restart file value should be for 20150405 00hr. Did you compare the SMC in restart file with the SMC in your continuous run for 2015040500 output file?

**Reply by PGhaneei** [reply]
Thank you for your quick response.
I started the simulations for both experiments from 2015040200 and am comparing the SMC states from the restart files 'RESTART.2015040500_DOMAIN1' for both experiments.

**Reply by cenlinhe** [reply]
I see. 
1. Did you see difference in other output variables (particularly those related to vegetation and carbon)? 
2. Did you see difference in SMC states for 2015040400 restart file and the 2015040400 output file?
3. Also, could you please do another test by using the dynamic_veg_option = 4 (currently you are using 6) to see if it still shows the difference in SMC states from the restart run and continuous run?

**Reply by PGhaneei** [reply]
Thank you for your reply.
1. Yes, for example the LAI states are different as well.

2. The soil moisture in the output file ('SOIL_M') and the soil moisture in the restart file ('SMC') are the same within each experiment, but they are different when comparing the two experiments.

3. I changed the dynamic_veg_option to 4, but the soil moisture values are still different between the two experiments.

Also, the differences become more as the simulation progresses over additional time steps.

I noticed that the HRLDAS setup file contains soil moisture and other initial states as well. Since I am using the same setup file, aren't the states be overwritten by the setup file?

**Reply by cenlinhe** [reply]
No, for restart run, the state variables will get their values from the restart file.
For two experiments: 
1. The continuous run is starting from 2015040200 and run for 8 days continously without stopping, turning of restart mode, but output restart file every 24 hours (you have files like RESTART.2014040300, RESTART.2014040400, RESTART.2014040500, ...).
2. The restart run will be starting from 2015040400 using the RESTART.2014040400 file generated by the first run (remember to uncomment the restart filename line in the namelist.hrldas and run it for a few days. Then compare output file for 2015040500.LDASOUT_DOMAIN files between the first and second runs.
Is this how you do your two experiments?

Also, could you please test by turning off the crop_option (i.e., use crop_option = 0) to see if it still shows the difference?

**Reply by PGhaneei** [reply]
Yes, this is how I set up my two experiments. 
I tested it with crop_option = 0, and that solved the problem. 
Would you please let me know what processes will be changed and which states or fluxes will be affected by setting crop_option = 0?
Thank you very much for your time and suggestions!

**Reply by cenlinhe** [reply, marked as answer]
setting crop_option=0 deactivates the dynamic crop growth process. Unless you specifically focus on crop field modeling, we do not need to activate this. This issue caused by crop_option may be because some of the crop-related state variables are missing in restart file. @CharlesZheZhang We may need to do a through check on this.

**Reply by PGhaneei** [reply]
Thanks for the information.
